### PR TITLE
Add option to save best model during training, save config json and model into run folder 

### PR DESCRIPTION
**Pull request by kyrylogr**

### CODE REVIEW

File: training/train_coco.py [hunk 1]
```diff
@@ -211,6 +204,16 @@ def train(model_conf, train_conf, data_conf):
             )
             train_loss_history.append(train_validation_loss)
             val_loss_history.append(val_validation_loss)
+            if val_validation_loss < best_val_loss:
```
Comment (bulygin1985) [inline training/train_coco.py:207]: It is better to use Callback like https://pytorch-lightning.readthedocs.io/en/1.5.10/extensions/generated/pytorch_lightning.callbacks.ModelCheckpoint.html . It could be a lot of parameters for BestChaeckpoint Saver: 1)"monitor" - to choose  "val_loss", "MAP", "F1" or other metrics as criteria to save 2) save weights only or the whole dict_states with optimizer, lr_schedule, etc 3) to save best N epoch , etc.  I think it's best to start with some minimalistic implementation, but with a possibility to extend it in future.

**Review by bulygin1985 — CHANGES_REQUESTED** [review]

File: training/train_coco.py (hunk 1 above)
Comment (kyrylogr) [reply training/train_coco.py:215]: I added callback with code to save only weights and best or current model.

File: callbacks/save_checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                and value >= self.best_metric_value
+                or not self.best_mode_is_min
+                and value <= self.best_metric_value
+            ):
+                return
+            self.best_metric_value = value
+            filename = self.filename.format(
+                epoch=epoch, metric=self.monitor_metric, value=value
+            )
+        filepath = os.path.join(self.log_dir, filename)
```
Comment (bulygin1985) [inline callbacks/save_checkpoint.py:56]: It is usefull message:"metric {...} is improved from val1 to val2"

**Review by bulygin1985 — APPROVED** [review]
